primefaces/src/main/java/org/primefaces/component/api/UIMediaBase.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2009-2025 PrimeTek Informatics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.component.api;
+
+import org.primefaces.cdk.api.Property;
+import org.primefaces.util.DynamicContentSrcBuilder;
+import org.primefaces.util.Lazy;
+
+import java.io.IOException;
+
+import jakarta.faces.component.UIComponentBase;
+import jakarta.faces.context.FacesContext;
+
+/**
+ * Base class for HTML5 audio and video components.
+ */
+public abstract class UIMediaBase extends UIComponentBase {
+
+    public String resolveSource(FacesContext context, UIMediaBase media) throws IOException {
```

Comment:
we dont need UIMediaBase media as param, juse use "this"

Reply:
Done